cli/Dockerfile

```diff
@@ -0,0 +1,28 @@
+# mcr.microsoft.com/oss/go/microsoft/golang:1.23.1-cbl-mariner2.0
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/oss/go/microsoft/golang@sha256:dcd95cadab21a855894c599c9e26bfb2179aa08e7d7f99c0e8546167712ba6ef AS builder
```

Comment:
I think the latest available digest is `sha256:8ac5373de7fde69d08c52e4a1ba40e976a543b3e93196f7fe07c3f91853865f3`

Reply:
I copied the digest from retina-agent dockerfile. I think it's ok to update the builder image as the latest for kubectl-retina only, but let all these base image sha updated when necessary in one shot.
https://github.com/microsoft/retina/blob/main/controller/Dockerfile#L6

Reply:
done